src/Spectre.libGmmModel/GaussianMixtureModel.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * GaussianMixtureModel.h
+ * Provides implementation of Gaussian Mixture Model return type.
+ *
+   Copyright 2017 Michal Gallus
+   
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+   
+   http://www.apache.org/licenses/LICENSE-2.0
+   
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+#include <gsl.h>
+
+namespace Spectre::libGmmModel
+{
+	/// <summary>
+	/// Represents a single component of a Gaussian Mixture
+	/// </summary>
+	struct GaussianComponent
+	{
+		/// <summary>
+		/// Mean value of the component or peak's location
+		/// </summary>
+		double mean;
+
+		/// <summary>
+		/// Standard deviation of the component or peak's width
+		/// </summary>
+		double deviation;
+
+		/// <summary>
+		/// Weight of the component or peak's height
+		/// </summary>
+		double weight;
+	};
+
+	/// <summary>
+	/// Represents a composite of gaussian distribution components 
+	/// that build up a Gaussian Mixture Model.
+	/// </summary>
+	struct GaussianMixtureModel
+	{
+		/// <summary>
+		/// Collection of Gaussian components
+		/// </summary>
+		gsl::span<GaussianComponent> components;
```

Comment:
This is all fine, glowing and shiny, but are we providing any possibility `GaussianMixtureModel` will *own* `GaussianComponent` instances? Maybe there should be a private e.g. `std::vector` field which will hold the data for the object lifespan, while `components` will just point to this?
**In general**: we want to *accept* `gsl::span`, to provide flexibility of *input*, while it is usually useful to have quite specific *output*. That's the concept I see in using `gsl::span` - flexibility of input, and well-specified output.